### PR TITLE
Add OpenFaaS chart 4.4.0

### PR DESCRIPTION
**Pull request by alexellis**
Adds OpenFaaS at chart version 4.4.0

* Corrected a typo in the CONTRIBUTING guide
* Added OpenFaaS namespaces as part of install (there are 2x needed)
* Corrected error for checking for rollout-status of deployment, it's not "openfaas", but "gateway" that needs to be checked for
* Updated the instructions for OpenFaaS to include a `$HELM_CHART_REPO`, given that a separate repo is maintained for the project.
* Tested with DOKS cluster in Amsterdam.

Signed-off-by: Alex Ellis <alexellis2@gmail.com>

Fixes: https://github.com/openfaas/faas/issues/1261
Fixes: https://github.com/digitalocean/marketplace-kubernetes/issues/16

### CODE REVIEW

**Comment by alexellis** [comment]
Deploying with the test script:

```
$ stacks/openfaas/deploy-local.sh 
namespace/openfaas created
namespace/openfaas-fn created
secret/basic-auth created
Context "do-nyc1-marketplace-of-1" modified.
configmap/alertmanager-config created
configmap/prometheus-config created
serviceaccount/openfaas-controller created
role.rbac.authorization.k8s.io/openfaas-controller created
rolebinding.rbac.authorization.k8s.io/openfaas-controller created
serviceaccount/openfaas-prometheus created
role.rbac.authorization.k8s.io/openfaas-prometheus created
rolebinding.rbac.authorization.k8s.io/openfaas-prometheus created
service/alertmanager created
service/basic-auth-plugin created
service/gateway-external created
service/gateway created
service/nats created
service/prometheus created
deployment.apps/alertmanager created
deployment.apps/basic-auth-plugin created
deployment.apps/faas-idler created
deployment.apps/gateway created
deployment.apps/nats created
deployment.apps/prometheus created
deployment.apps/queue-worker created
Waiting for deployment "gateway" rollout to finish: 0 of 1 updated replicas are available...
```

Checking it works:

```
kubectl get secret -n openfaas basic-auth -o jsonpath="{.data.basic-auth-password}" | base64 --decode > password.txt

export OPENFAAS_URL=$(kubectl get svc -n openfaas gateway-external -o  jsonpath='{.status.loadBalancer.ingress[*].ip}'):8080

echo "Your gateway URL is: ${OPENFAAS_URL}"

cat password.txt | faas-cli login --username admin --password-stdin

faas-cli store deploy nodeinfo

faas-cli list -v

faas-cli describe nodeinfo
```

**Comment by arosales** [comment]
@alexellis thanks for this submission.  I was glad to see a pull request for OpenFaas. 👍
We'll take a look on Monday when more folks are back at their keyboards.

File: src/openfaas/4.4.0/values.yamle [hunk 1]
```diff
@@ -0,0 +1,139 @@
+functionNamespace: openfaas-fn
```
Comment (keladhruv) [inline src/openfaas/4.4.0/values.yamle]: Is the file name supposed to be values.yamle?

Comment (alexellis) [reply src/openfaas/4.4.0/values.yamle]: It's generated by `sed`. I don't see it as a blocker. Would like to try and get this in for additional e2e testing within DOKS.

File: stacks/openfaas/deploy-local.sh [hunk 2]
```diff
@@ -0,0 +1,23 @@
+#!/bin/sh
+
+set -e
+
+ROOT_DIR=$(git rev-parse --show-toplevel)
+
+kubectl apply -f https://raw.githubusercontent.com/openfaas/faas-netes/master/namespaces.yml
```
Comment (paulanunda) [inline stacks/openfaas/deploy-local.sh]: Mind including the namespace manifest inline? ie:

```
cat <<EOF | kubectl apply -f -
apiVersion: v1
kind: Namespace
metadata:
  name: openfaas
  labels:
    role: openfaas-system
    access: openfaas-system
    istio-injection: enabled
---
apiVersion: v1
kind: Namespace
metadata:
  name: openfaas-fn
  labels:
    istio-injection: enabled
    role: openfaas-fn
EOF
```

File: stacks/openfaas/deploy.sh [hunk 3]
```diff
@@ -0,0 +1,21 @@
+#!/bin/sh
+
+set -e
+
+kubectl apply -f https://raw.githubusercontent.com/openfaas/faas-netes/master/namespaces.yml
```
Comment (paulanunda) [inline stacks/openfaas/deploy.sh]: Mind including the namespace manifest inline? ie:

```
cat <<EOF | kubectl apply -f -
apiVersion: v1
kind: Namespace
metadata:
  name: openfaas
  labels:
    role: openfaas-system
    access: openfaas-system
    istio-injection: enabled
---
apiVersion: v1
kind: Namespace
metadata:
  name: openfaas-fn
  labels:
    istio-injection: enabled
    role: openfaas-fn
EOF
```

Comment (alexellis) [reply stacks/openfaas/deploy.sh]: I would rather track it from the main repo because it would better match the helm install instructions. Can that work?

Comment (alexellis) [reply stacks/openfaas/deploy.sh]: If it's not workable, let me know and I can hard-code the namespace into the shell scripts.

Comment (paulanunda) [reply stacks/openfaas/deploy.sh]: Given that it matches the Helm install instructions, the current implementation looks good to me.

**Comment by alexellis** [comment]
Thank you!

**Pull request merged**